### PR TITLE
JAMES-3772 Close ReactorRabbitMQChannelPool synchronously

### PR DESCRIPTION
**Pull request by chibenwa**
Fixes:

```
org.apache.james.backends.rabbitmq.ReactorRabbitMQChannelPoolTest.notUsedChannelShouldBeClosedWhenPoolIsClosed
Error Message

Expecting value to be false but was true

Stacktrace

org.opentest4j.AssertionFailedError: 

Expecting value to be false but was true
	at org.apache.james.backends.rabbitmq.ReactorRabbitMQChannelPoolTest.notUsedChannelShouldBeClosedWhenPoolIsClosed(ReactorRabbitMQChannelPoolTest.java:117)
```

### CODE REVIEW

File: backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/ReactorRabbitMQChannelPool.java [hunk 1]
```diff
@@ -796,10 +799,9 @@ public Mono<Void> createWorkQueue(QueueSpecification queueSpecification) {
     @Override
     public void close() {
         sender.close();
-        ImmutableList.copyOf(refs.values())
-            .stream()
-            .map(PooledRef::poolable)
-            .forEach(channel -> getChannelCloseHandler().accept(SignalType.ON_NEXT, channel));
+       Flux.fromIterable(refs.values())
+           .flatMap(PooledRef::invalidate)
+           .blockLast();
```
Comment (vttranlina) [inline backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/ReactorRabbitMQChannelPool.java:802]: Should we use .subscribe(); ?

Comment (chibenwa) [reply backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/ReactorRabbitMQChannelPool.java:802]: No here we want to be sure that it is finished.

Also stopping the app is not a reactive pipeline: blocking is OK.

**Review by Arsnael — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]

**Review by vttranlina — APPROVED** [review]

**Comment by vttranlina** [comment]
Almost all ci error from Cassandra docker, maybe it is a temporary error, we should re-run ci?

**Comment by Arsnael** [comment]
@vttranlina The CI is green... are you sure? :)

**Comment by vttranlina** [comment]
> @vttranlina The CI is green... are you sure? :)

40 minutes before, it is red. 
ignore my above comment

**Pull request merged**